src/gui/qgsadvanceddigitizingdockwidget.cpp

```diff
@@ -555,6 +555,8 @@ bool QgsAdvancedDigitizingDockWidget::applyConstraints( QgsMapMouseEvent *e )
 
   mSnapMatch = context.snappingUtils->snapToMap( point );
 
+  mSnappedToVertex = mSnapMatch.hasVertex();
```

Comment:
wouldn't it be safer to remove this unused member and rather adapt the method `snappedToVertex()` to return `mSnapMatch.isValid() && mSnapMatch.hasVertex()` ?

Reply:
@3nids right. Done. Thanks